lib/js_of_ocaml/dom_html.ml

```diff
@@ -2165,6 +2165,37 @@ class type history =
 
 class type undoManager = object end
 
+class type shareOption =
+  object
+    method url : js_string t optdef prop
+
+    method text : js_string t optdef prop
+
+    method title : js_string t optdef prop
+  end
+
+class type promise =
+  object
+    method _then : ('self t, event t -> unit) Js.meth_callback opt -> promise t meth
+
+    method catch : ('self t, js_string -> unit) Js.meth_callback -> unit meth
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share says  the promise can be rejected with a DomException not a string.